clients/clients.go

```diff
@@ -341,47 +335,106 @@ func (c *Clients) onRoomMemberEvent(client *mautrix.Client, event *mevt.Event) {
 	}
 }
 
-func (c *Clients) newClient(config api.ClientConfig) (*mautrix.Client, error) {
+func (c *Clients) initClient(botClient *BotClient) error {
+	config := botClient.config
 	client, err := mautrix.NewClient(config.HomeserverURL, config.UserID, config.AccessToken)
 	if err != nil {
-		return nil, err
+		return err
 	}
+
 	client.Client = c.httpClient
+	client.DeviceID = config.DeviceID
```

Comment:
What happens if no device ID is provided for the client? We should support this (and just not have E2E work).

Reply:
That's what happens - the device ID is only used for e2ee operations. I added a warning to let the user know they didn't specify a device ID, other than that unencrypted rooms should work normally.